sos/plugins/crash_report.py

```diff
@@ -0,0 +1,249 @@
+                    return 0
+        elif file_type == "vmlinu":
+            for filename in find("vmlinu*", start_path):
+                """
+                in rhel : 3.10.0-327.el7.ppc64/vmlinux
+                in ubuntu: vmlinux-4.4.0-21-generic
+                hence below check confirms that vmlinux is of crashed kernel
+                """
+                if filename.find(file_opt) != -1:
+                    self.vmlinux_path = filename
+                    return 0
+        return -1
+
+    def get_crash_time(self, line):
+        crashing_time = (line.split()[4] + ' ' + line.split()[5] +
+                         ' ' + line.split()[6])
+        try:
+            t2 = datetime.strptime(crashing_time, "%b %d %H:%M")
+        except ValueError:
+            t2 = None
+            crashing_time = (line.split()[3] + ' ' + line.split()[4] +
+                             ' ' + line.split()[5])
+            try:
+                t2 = datetime.strptime(crashing_time, "%b %d %H:%M")
+            except ValueError:
+                t2 = None
+                return ""
+        return crashing_time
+
+    def get_vmlinux_vmcore_path(self):
+        # retrieve data from last command
+        last_cmd_res = sos_get_command_output('last')
+        if last_cmd_res['status']:
+            self.message_to_log = '\n\n' + 'last cmd failed' + '\n\n'
+            self.dump_log_data()
+            return -1
+
+        crash_found = False
+
+        for line in last_cmd_res['output'].splitlines():
+            line = line.rstrip()
+            if crash_found is True:
+                reboot_str = (line.split()[0] + ' ' + line.split()[1] + ' ' +
+                              line.split()[2])
+                if reboot_str == 'reboot system boot':
+                    line_contain_os = line
+                    self.crashed_kernel_name = line.split()[3]
+                    break
+            elif "crash" in line:
+                line_contain_crash = line
+                crash_found = True
+        if not crash_found or self.crashed_kernel_name == "":
+            self.message_to_log = (
+             '\n\n' + 'Failed in retrieving crash from last cmd output:' +
+             '\n\n')
+            return -1
+        crashing_time = self.get_crash_time(line_contain_crash)
+        if not crashing_time:
+            self.message_to_log = (
+             '\n\n' + 'Failed in retrieving time for last crash:' +
+             '[ ' + line_contain_crash + ' ]' + '\n\n')
+            return -1
+        ret = self.retrieve_path(self.dbg_vmlinux_path, "vmlinu",
+                                 self.crashed_kernel_name, crashing_time)
+        if ret:
+            self.message_to_log = (
+             '\n\n' + 'Failed in retrieving debug linux:' +
+             '[ ' + self.crashed_kernel_name + ' ]' + ' inside path: ' +
+             '[ ' + self.dbg_vmlinux_path + ' ]' +
+             '\nPlease install debug linux and rerun sosreport\n\n')
```

Comment:
Also this requires the user to install something else on their system to retrieve this data. sosreport should not halt if it can't find the debug symbols and continue on.